src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -370,26 +378,36 @@ def isWorkflowCompletedWithLogCollectAndCleanUp(self, requestName):
         reqInfoInstance = RequestInfo(requestInfo[requestName])
         return reqInfoInstance.isWorkflowFinished()
 
-    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1):
-
+    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1, usePycurl=True):
+        reqStart = time.time()
         options = {'reduce': True, 'group_level': 5, 'startkey': [requestName],
                    'endkey': [requestName, {}]}
         results = self._getCouchView("jobsByStatusWorkflow", options)
+
         jobDetails = {}
-        for row in results['rows']:
-            # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
-            startKey = row["key"][:4]
-            endKey = []
-            site = row["key"][4]
-            if site:
-                startKey.append(site)
-
-            endKey.extend(startKey)
-            endKey.append({})
-            numOfError = row["value"]
-
-            jobInfo = self.jobDetailByTasks(startKey, endKey, numOfError, sampleSize)
+        if usePycurl is True:
+            jobInfo = getTaskJobSummaryByRequestPycurl(results, sampleSize, self.serviceOpts)
             jobDetails = nestedDictUpdate(jobDetails, jobInfo)
+        else:
+            # then it is sequential
+            for row in results['rows']:
+                # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
+                startKey = row["key"][:4]
+                endKey = []
+                site = row["key"][4]
+                if site:
```

Comment:
Would it make sense to use try-except in place of the if statement?

Reply:
I think it can be either None or an empty string. In such cases, we do not want to have it as part of the query, hence we don't append it.